src/Sylius/Bundle/ApiBundle/spec/CommandHandler/Checkout/AddressOrderHandlerSpec.php

```diff
@@ -133,6 +139,49 @@ function it_handles_addressing_an_order_for_logged_in_shop_user(
         $stateMachine->can('address')->willReturn(true);
         $stateMachine->apply('address')->shouldBeCalled();
 
+        $manager->persist($order)->shouldBeCalled();
+        $manager->flush()->shouldBeCalled();
+
+        $this($addressOrder);
+    }
+
+    function it_handles_addressing_an_order_for_not_logged_in_shop_user(
+        OrderRepositoryInterface $orderRepository,
+        CustomerRepositoryInterface $customerRepository,
+        FactoryInterface $customerFactory,
+        ObjectManager $manager,
+        StateMachineFactoryInterface $stateMachineFactory,
+        CustomerInterface $customer,
+        AddressInterface $billingAddress,
+        AddressInterface $shippingAddress,
+        OrderInterface $order,
+        StateMachineInterface $stateMachine
+    ): void {
+        $addressOrder = new AddressOrder(
+            'r2d2@droid.com',
+            $billingAddress->getWrappedObject(),
+            $shippingAddress->getWrappedObject()
+        );
+        $addressOrder->setOrderTokenValue('ORDERTOKEN');
+
+        $orderRepository->findOneBy(['tokenValue' => 'ORDERTOKEN'])->willReturn($order);
+
+        $order->getCustomer()->willReturn(null);
+
+        $customerRepository->findOneBy(['email' => 'r2d2@droid.com'])->willReturn($customer);
+
+        $order->getCustomer()->shouldBeCalled();
+        $order->setCustomer($customer)->shouldBeCalled();
+        $order->setBillingAddress($billingAddress)->shouldBeCalled();
+        $order->setShippingAddress($shippingAddress)->shouldBeCalled();
+
+        $stateMachineFactory->get($order, OrderCheckoutTransitions::GRAPH)->willReturn($stateMachine);
+        $stateMachine->can('address')->willReturn(true);
```

Comment:
can you use here this const value?
```suggestion
        $stateMachine->can(OrderCheckoutTransitions::TRANSITION_ADDRESS)->willReturn(true);
```